app/src/main/java/com/example/geoquiz/MainActivity.kt

```diff
@@ -84,18 +79,29 @@ class MainActivity : AppCompatActivity() {
         savedInstanceState.putInt(KEY_CURRENT_INDEX, quizViewModel.currentIndex)
     }
 
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        super.onActivityResult(requestCode, resultCode, data)
-        if (resultCode != Activity.RESULT_OK) {
-            return
-        }
-        if (requestCode == REQUEST_CODE_CHEAT) {
-            quizViewModel.isCheater = data?.getBooleanExtra(EXTRA_ANSWER_SHOWN, false) ?: false
+    private fun onActivityResult(result: ActivityResult) {
+        if (result.resultCode == RESULT_OK) {
+            val intent = result.data
+            quizViewModel.isCheater =
+                intent?.getBooleanExtra(CheatActivity.EXTRA_ANSWER_SHOWN, false) ?: false
         }
     }
 
     private fun restoreState(savedState: Bundle?) {
         val currentIndex = savedState?.getInt(KEY_CURRENT_INDEX, 0) ?: 0
         quizViewModel.currentIndex = currentIndex
     }
+
+    private fun bindViews() {
+        trueButton = findViewById(R.id.true_button)
+        falseButton = findViewById(R.id.false_button)
+        nextButton = findViewById(R.id.next_button)
+        cheatButton = findViewById(R.id.cheat_button)
+        questionTextView = findViewById(R.id.question_text_view)
+    }
+
+    companion object {
+        private const val TAG = "MainActivity"
+        private const val KEY_CURRENT_INDEX = "index"
```

Comment:
Usually the content inside the key matches the naming of the key, so in this case "index" -> "current_index"